tawazi/dag.py

```diff
@@ -506,37 +371,64 @@ def _get_leaves_ids(
         Returns:
             List[IdentityHash]: ExecNodes' Identities
         """
-        leaves_ids = []
-        for tag_or_id_or_node in twz_nodes:
-            if isinstance(tag_or_id_or_node, ExecNode):
-                leaves_ids.append(tag_or_id_or_node.id)
-            # todo: do further validation for the case of the tag!!
-            elif isinstance(tag_or_id_or_node, (IdentityHash, tuple)):
-                tag_or_id = tag_or_id_or_node
-
-                # if leaves_identification is not ExecNode, it can be either
-                #  1. a Tag (Highest priority in case an id with the same value exists)
-                if node := self.tag_node_dict.get(tag_or_id):
-                    leaves_ids.append(node.id)
-                #  2. or a node id!
-                elif isinstance(tag_or_id, IdentityHash):
-                    node = self.get_node_by_id(tag_or_id)
-                    leaves_ids.append(node.id)
+        if twz_nodes is None:
+            # TODO: make cached!
+            leaves_ids = [xn.id for xn in self.exec_nodes]
+        # 2. create leaves_ids
+        else:
+            leaves_ids = []
+            for tag_or_id_or_node in twz_nodes:
+                if isinstance(tag_or_id_or_node, ExecNode):
+                    leaves_ids.append(tag_or_id_or_node.id)
+                # todo: do further validation for the case of the tag!!
+                elif isinstance(tag_or_id_or_node, (IdentityHash, tuple)):
+                    tag_or_id = tag_or_id_or_node
```

Comment:
is reassigning a variable useful since we don't mutate `tag_or_id` ?

Reply:
Nope this is only for clarification purposes

Reply:
okok, do you want to leave it then ?

Reply:
ok